package/src/components/MessageList/MessageList.tsx

```diff
@@ -367,11 +367,9 @@ const MessageListWithContext = <
     ) => message && lastRead && message.created_at && lastRead < message.created_at,
   );
 
-  const channelLastReadRef = useRef(channel?.initialized ? channel.lastRead() : undefined);
-
-  useEffect(() => {
-    channelLastReadRef.current = channel?.initialized ? channel.lastRead() : undefined;
-  }, [channel]);
+  // ref for channel to use in useEffect without triggering it on channel change
+  const channelRef = useRef(channel);
+  channelRef.current = channel;
```

Comment:
Out of curiosity, if we are already setting the `channel` as an argument to `useRef`, why do we need to set `channelRef.current` again?

Reply:
From what I know,

```
  const channelRef = useRef(channel);
  channelRef.current = channel;
```

is same as saying:

```
  const channelRef = useRef(channel);
```

Please help me if I am missing out on some fundamentals.

Reply:
No it is not.. `channelRef.current = channel;` ensures that ref is changed on everyrender

Reply:
useRef(channel) only sets on initial render